autonomi/src/client/data_types/chunk.rs

```diff
@@ -113,10 +109,10 @@ fn hash_to_short_string(input: &str) -> String {
 
 impl Client {
     /// Get a chunk from the network.
-    pub async fn chunk_get(&self, addr: ChunkAddr) -> Result<Chunk, GetError> {
+    pub async fn chunk_get(&self, addr: ChunkAddress) -> Result<Chunk, GetError> {
```

Comment:
Pointers point to `ChunkAddress` which is why we need the chunk API to return chunk addresses.